lib/jekyll/theme.rb

```diff
@@ -43,7 +47,7 @@ def path_for(folder)
     end
 
     def realpath_for(folder)
-      File.realpath(Jekyll.sanitized_path(root, "_#{folder}"))
+      File.realpath(Jekyll.sanitized_path(root, folder.to_s))
```

Comment:
I'm torn here. Do we want to pass vanilla symbols (without the `_`) and special case `assets` so that we treat all of them as named folders, rather than paths?